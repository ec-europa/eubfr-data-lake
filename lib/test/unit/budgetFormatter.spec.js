/**
 * @jest-environment node
 */

const { sanitizeBudgetItem } = require('../../budgetFormatter');

describe('budgetFormatter module: sanitizeBudgetItem helper', () => {
  test('Should return default BudgetItem type when no arguments', () => {
    const result = sanitizeBudgetItem();
    expect(result).toMatchSnapshot();
  });

  test('Currency ECU should be treated equal to EUR', () => {
    const input = { value: '1259101', currency: 'ECU', raw: 'ECU 1 259 101' };
    const result = sanitizeBudgetItem(input);
    expect(result).toMatchSnapshot();
    expect(result.currency).toBe('EUR');

    const valueFromRaw = result.raw.replace(/ECU/gi, '').replace(/\s/g, '');
    expect(result.value).toEqual(Number(valueFromRaw));
  });
});
