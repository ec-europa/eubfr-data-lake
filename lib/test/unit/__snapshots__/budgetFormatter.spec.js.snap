// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`budgetFormatter module: sanitizeBudgetItem helper Currency ECU should be treated equal to EUR 1`] = `
Object {
  "currency": "EUR",
  "raw": "ECU 1 259 101",
  "value": 1259101,
}
`;

exports[`budgetFormatter module: sanitizeBudgetItem helper Should return default BudgetItem type when no arguments 1`] = `
Object {
  "currency": "",
  "raw": "",
  "value": 0,
}
`;
